#Date: April 30, 2024
#Subject: [PATCH] Patch to fix compilation warning
#		 in libfyaml-master directory. 
#
#Source: Uses code from the libfyaml-master 
#Signed-off-by: kanjoe24 <165808281+kanjoe24@users.noreply.github.com>
#---
#
#==========================================================================================
diff -r -c libfyaml-master/src/lib/fy-composer.h libfyaml-master-working/src/lib/fy-composer.h
*** libfyaml-master/src/lib/fy-composer.h	2024-04-22 17:20:09.000000000 +0000
--- libfyaml-master-working/src/lib/fy-composer.h	2024-04-26 14:33:21.000000000 +0000
***************
*** 48,54 ****
  
  struct fy_composer *fy_composer_create(struct fy_composer_cfg *cfg);
  void fy_composer_destroy(struct fy_composer *fyc);
! int fy_composer_process_event(struct fy_composer *fyc, struct fy_event *fye);
  
  struct fy_composer_cfg *fy_composer_get_cfg(struct fy_composer *fyc);
  void *fy_composer_get_cfg_userdata(struct fy_composer *fyc);
--- 48,54 ----
  
  struct fy_composer *fy_composer_create(struct fy_composer_cfg *cfg);
  void fy_composer_destroy(struct fy_composer *fyc);
! enum fy_composer_return fy_composer_process_event(struct fy_composer *fyc, struct fy_event *fye);
  
  struct fy_composer_cfg *fy_composer_get_cfg(struct fy_composer *fyc);
  void *fy_composer_get_cfg_userdata(struct fy_composer *fyc);
diff -r -c libfyaml-master/src/lib/fy-diag.c libfyaml-master-working/src/lib/fy-diag.c
*** libfyaml-master/src/lib/fy-diag.c	2024-04-22 17:20:09.000000000 +0000
--- libfyaml-master-working/src/lib/fy-diag.c	2024-04-30 14:17:09.000000000 +0000
***************
*** 21,26 ****
--- 21,27 ----
  #include <ctype.h>
  
  #include <libfyaml.h>
+ #include <gnu_lib_alloc_strings.h>
  
  #include "fy-parse.h"
  
diff -r -c libfyaml-master/src/lib/fy-doc.c libfyaml-master-working/src/lib/fy-doc.c
*** libfyaml-master/src/lib/fy-doc.c	2024-04-22 17:20:09.000000000 +0000
--- libfyaml-master-working/src/lib/fy-doc.c	2024-04-30 14:14:21.000000000 +0000
***************
*** 20,25 ****
--- 20,27 ----
  
  #include <libfyaml.h>
  
+ #include <gnu_lib_alloc_strings.h>
+ 
  #include "fy-parse.h"
  #include "fy-doc.h"
  
diff -r -c libfyaml-master/src/lib/fy-event.c libfyaml-master-working/src/lib/fy-event.c
*** libfyaml-master/src/lib/fy-event.c	2024-04-22 17:20:09.000000000 +0000
--- libfyaml-master-working/src/lib/fy-event.c	2024-04-30 14:15:28.000000000 +0000
***************
*** 19,24 ****
--- 19,25 ----
  #include <stdarg.h>
  
  #include <libfyaml.h>
+ #include <gnu_lib_alloc_strings.h>
  
  #include "fy-parse.h"
  #include "fy-emit.h"
diff -r -c libfyaml-master/src/lib/fy-input.c libfyaml-master-working/src/lib/fy-input.c
*** libfyaml-master/src/lib/fy-input.c	2024-04-22 17:20:09.000000000 +0000
--- libfyaml-master-working/src/lib/fy-input.c	2024-04-30 14:18:06.000000000 +0000
***************
*** 24,29 ****
--- 24,30 ----
  #include <errno.h>
  
  #include <libfyaml.h>
+ #include <gnu_lib_alloc_strings.h>
  
  #include "fy-parse.h"
  #include "fy-ctype.h"
diff -r -c libfyaml-master/src/lib/fy-parse.c libfyaml-master-working/src/lib/fy-parse.c
*** libfyaml-master/src/lib/fy-parse.c	2024-04-22 17:20:09.000000000 +0000
--- libfyaml-master-working/src/lib/fy-parse.c	2024-04-26 14:33:21.000000000 +0000
***************
*** 34,39 ****
--- 34,40 ----
  #define ATOM_SIZE_CHECK
  #endif
  
+ #if 0
  const char *fy_library_version(void)
  {
  #ifndef VERSION
***************
*** 43,48 ****
--- 44,50 ----
  	return VERSION;
  #endif
  }
+ #endif
  
  int fy_parse_input_append(struct fy_parser *fyp, const struct fy_input_cfg *fyic)
  {
diff -r -c libfyaml-master/src/lib/fy-token.c libfyaml-master-working/src/lib/fy-token.c
*** libfyaml-master/src/lib/fy-token.c	2024-04-22 17:20:09.000000000 +0000
--- libfyaml-master-working/src/lib/fy-token.c	2024-04-30 14:17:44.000000000 +0000
***************
*** 19,24 ****
--- 19,25 ----
  #include <stdarg.h>
  
  #include <libfyaml.h>
+ #include <gnu_lib_alloc_strings.h>
  
  #include "fy-parse.h"
  
diff -r -c libfyaml-master/src/lib/fy-walk.c libfyaml-master-working/src/lib/fy-walk.c
*** libfyaml-master/src/lib/fy-walk.c	2024-04-22 17:20:09.000000000 +0000
--- libfyaml-master-working/src/lib/fy-walk.c	2024-04-30 14:16:21.000000000 +0000
***************
*** 20,25 ****
--- 20,26 ----
  #include <limits.h>
  
  #include <libfyaml.h>
+ #include <gnu_lib_alloc_strings.h>
  
  #include "fy-parse.h"
  #include "fy-doc.h"
